src/settings.cpp

```diff
@@ -742,6 +702,22 @@ void StringSettingDesc::FormatValue(char *buf, const char *last, const void *obj
 	}
 }
 
+bool StringSettingDesc::SameValue(const IniItem *item, void *object) const
+{
+	/* The ini parsing removes the quotes, which are needed to retain the spaces in STRQs,
+	 * so those values are always different in the parsed ini item than they should be. */
+	if (GetVarMemType(this->save.conv) == SLE_VAR_STRQ) return false;
+
+	const std::string &str = this->Read(object);
+	return item->value->compare(str) == 0;
+}
+
+bool ListSettingDesc::SameValue(const IniItem *item, void *object) const
+{
+	/* Checking for equality is way more expensive than just writing the value. */
```

Comment:
This heavily assumes a usage of this function already, which might give future-us a wtf moment. I suggest we document clearly in the doxygen that it is an approximation or something .. no clue how to word that correctly :) But at least that it is clear it might return `false` even if they are equal.